### PR TITLE
Rte translate button + translation via xml transformation

### PR DESCRIPTION
**Pull request by VP-DS**
https://github.com/vivid-planet/comet/pull/1587 New Pull Request using comet repo branch instead of forked one

### CODE REVIEW

**Comment by VP-DS** [comment]
There is no good way to disable the translation button like for the FinalFormInput with adding a hideTranslate props.
There is SupportedThings on the rte but that naming would not match an hideTranslate option.

**Comment by nsams** [comment]
is this xml transformation really necessary? that is very complaex and possibly incomplete

**Comment by VP-DS** [comment]
> is this xml transformation really necessary? that is very complaex and possibly incomplete
It is necessary to keep the knowledge of which words have inline stylings after the translation.

(The xml code was copied from https://github.com/vivid-planet/comet/pull/1040/files)

I tried some draftJSToHtml + htmlToDraftJS package as an alternative, but there you lost subscript and superscript.

File: packages/admin/admin-rte/src/core/translation/xml/transformStateToXml.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    INLINE_STYLE.SUP,
+];
+
+const DEFAULT_STYLE_MAP = {
+    [INLINE_STYLE.BOLD]: { element: "inline" },
+    [INLINE_STYLE.ITALIC]: { element: "inline" },
+    [INLINE_STYLE.STRIKETHROUGH]: { element: "inline" },
+    [INLINE_STYLE.UNDERLINE]: { element: "inline" },
+    [INLINE_STYLE.SUB]: { element: "inline" },
+    [INLINE_STYLE.SUP]: { element: "inline" },
```
Comment (thomasdax98) [inline packages/admin/admin-rte/src/core/translation/xml/transformStateToXml.ts:42]: Is this flexible enough?

It's possible to add custom inline styles in the project afaik (@johnnyomair ?). So we would have to also support them.

**Review by thomasdax98 — COMMENTED** [review]
The whole XML logic looks very complicated. I didn't really review that as I think it's not possible for me to understand it completely without investing hours and testing it first hand.

I trust that you tested it thoroughly for different cases @VP-DS

**Comment by VP-DS** [comment]
> The whole XML logic looks very complicated. I didn't really review that as I think it's not possible for me to understand it completely without investing hours and testing it first hand.
> 
> I trust that you tested it thoroughly for different cases @VP-DS

@thomasdax98 I tested it by

File: packages/admin/admin-rte/src/core/translation/xml/transformStateToXml.ts (hunk 1 above)
Comment (thomasdax98) [reply packages/admin/admin-rte/src/core/translation/xml/transformStateToXml.ts:42]: see https://github.com/vivid-planet/comet/pull/1587/files#r1455283058

**Pull request closed without merging**